libs/components/data-manager/testing/src/modules/data-manager/data-manager-harness.spec.ts

```diff
@@ -0,0 +1,346 @@
+        toolbarHarness.getSection({ dataSkyId: 'toolbar-section' }),
+      ).toBeResolved();
+    });
+
+    it('should get arrays of data manager toolbar components', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      const primaryItems = await toolbarHarness.getPrimaryItems();
+      expect(primaryItems.length).toBe(1);
+
+      const leftItems = await toolbarHarness.getLeftItems();
+      expect(leftItems.length).toBe(2);
+
+      const rightItems = await toolbarHarness.getRightItems();
+      expect(rightItems.length).toBe(1);
+
+      const sections = await toolbarHarness.getSections();
+      expect(sections.length).toBe(1);
+    });
+
+    it('should throw an error if no toolbar components are found matching criteria', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      await expectAsync(
+        toolbarHarness.getPrimaryItems({ dataSkyId: 'other-item' }),
+      ).toBeRejectedWithError(
+        'Unable to find any data manager toolbar primary items with filter(s): {"dataSkyId":"other-item"}',
+      );
+      await expectAsync(
+        toolbarHarness.getLeftItems({ dataSkyId: 'other-item' }),
+      ).toBeRejectedWithError(
+        'Unable to find any data manager toolbar left items with filter(s): {"dataSkyId":"other-item"}',
+      );
+      await expectAsync(
+        toolbarHarness.getRightItems({ dataSkyId: 'other-item' }),
+      ).toBeRejectedWithError(
+        'Unable to find any data manager toolbar right items with filter(s): {"dataSkyId":"other-item"}',
+      );
+      await expectAsync(
+        toolbarHarness.getSections({ dataSkyId: 'other-section' }),
+      ).toBeRejectedWithError(
+        'Unable to find any data manager toolbar sections with filter(s): {"dataSkyId":"other-section"}',
+      );
+    });
+
+    it('should get the view actions', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      const viewActions = await toolbarHarness.getViewActions();
+      expect(viewActions).not.toBeNull();
+    });
+
+    it('should get the filter button', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      const filterButtonHarness = await toolbarHarness.getFilterButton();
+      expect(filterButtonHarness).not.toBeNull();
+    });
+
+    it('should get the sort button', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      const sortButtonHarness = await toolbarHarness.getSortButton();
+      expect(sortButtonHarness).not.toBeNull();
+    });
+
+    it('should get the search bar', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      const searchHarness = await toolbarHarness.getSearch();
+      expect(searchHarness).not.toBeNull();
+    });
+
+    it('should click the select all button', async () => {
+      const { toolbarHarness } = await setupToolbarTest();
+
+      await expectAsync(toolbarHarness.clickSelectAll()).toBeResolved();
```

Comment:
just a thought but could we have a function `getSelectedColumns`? Would that be beneficial for consumers?